File: algorithms/python/slowestKey/slowestKey.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/slowest-key/
# Author: Miao Zhang
# Date:   2021-05-24

class Solution:
    def slowestKey(self, releaseTimes: List[int], keysPressed: str) -> str:
        n = len(releaseTimes)
        c = ' '
        maxtime = 0
        for i in range(n):
            k = keysPressed[i]
            time = releaseTimes[i] - (releaseTimes[i - 1] if i > 0 else 0)
            if time > maxtime or (time == maxtime and k > c):
                maxtime = time
                c = k
        return c

File: algorithms/cpp/slowestKey/slowestKey.cpp
// Source: https://leetcode.com/problems/slowest-key/
// Author: Miao Zhang
// Date:   2021-05-24

class Solution {
public:
    char slowestKey(vector<int>& releaseTimes, string keysPressed) {
        int n = releaseTimes.size();
        char c = ' ';
        int maxtime = 0;
        for (int i = 0; i < n; i++) {
            char k = keysPressed[i];
            int time = releaseTimes[i] - (i > 0 ? releaseTimes[i - 1] : 0);
            if (maxtime < time || (maxtime == time && k > c)) {
                maxtime = time;
                c = k;
            }
        }
        return c;
    }
};
